fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -1271,4 +1277,55 @@ void continuations(boolean sorted) throws Exception {
             }
         }
     }
+
+    /**
+     * This tests is reproducing a thread deadlock that happened with Lucene threads when reading schema information.
+     * The fix for the deadlock has removed the {@code Map.computeIfAbsent} call (with the {@code synchronized} block)
+     * that blocked the {@code asyncToSync} calls.
+     * This test creates a large enough set of {@link org.apache.lucene.index.SegmentReader} futures so as to flood the thread pool,
+     * and then queries Lucene. THe large number of futures in the pool, together with the {@code synchronized} block
+     * would cause a deadlock.
+     */
+    @Test
+    void testQueryWithManyDocuments() {
+        String[] words = generateRandomWords(1000);
```

Comment:
IF `generateRandomWords` is `public`, and used by a variety of tests, it should probably be moved to `LuceneIndexTestutils`

Reply:
Done.